packages/wheelhouse-frontend/src/components/logs-data/LogsDataDisplay.js

```diff
@@ -18,32 +40,101 @@ class LogsDataDisplay extends Component {
     });
   }
 
+  addUpdatedLog() {
+    this.props.dispatch({
+      type: ADD_UPDATED_LOG
+    });
+  }
+
+  toggleIndivApp(appName) {
+    this.setState({
+      showLogs: {
+        ...this.state.showLogs, 
+        [appName]: !this.state.showLogs[appName]
+      }
+    });
+  }
+
+  toggleAllApps(bool) {
+    let showLogsCopy = Object.assign({}, this.state.showLogs);
+    for (let key in showLogsCopy) {
+      showLogsCopy[key] = bool; 
+    }
+    this.setState({
+      showLogs: showLogsCopy 
+    });
+  }
+
   render() {
     const { logs } = this.props;
     const lines = logs.map((line, idx) => {
+
       const time = logHandlers.timeConverter();
       const hashed = logHandlers.hashCode(line.appName);
-      const randomColor = logHandlers.intToRGB(hashed); 
-      const divStyle = {
+      const randomColor = logHandlers.intToRGB(hashed);
+
+      let blockOrNone;
+      this.state.showLogs[line.appName] ? blockOrNone = null : blockOrNone = "none";
+
+      const textColor = {
         color: randomColor
       };
+
+      const displayBlockOrNone = {
+        display: blockOrNone
+      };
+
       return (
         <LogLine
           key={idx} 
           timeStamp={time}
           appName={line.appName}
-          customStyle={divStyle}
+          color={textColor}
           serverStatus={line.serverStatus}
           expectedAction={line.expectedAction}
+          display={displayBlockOrNone}
         />
       );
     });
-  
+
+    const logButtons = () => {
+      const buttons = []; 
+
+      for (let key in this.state.showLogs) {
+        let buttonClass= !this.state.showLogs[key] ? "button-outline hide-log-button log-filter-button" : "log-filter-button";
+        buttons.push(
+            <li key={key}>
+              <button 
+                onClick={() => {this.toggleIndivApp(key); }}
+                className={buttonClass}>
+                {key}
+              </button>
+            </li>
+          ); 
+      }
+      return buttons;
+    };
+    //log-filter-container should be it's own reusable component, but it loses the ability to call
+    //logButtons when it's being called as a prop
     return (
       <div className="container">
         <div className="row">
-          <div className="sidebar-container"><Sidebar /></div>
-          <div className="content-container logs-container">{lines}</div>
+          <div className="sidebar-container">
+            <Sidebar />
+            <div className="log-filter-container">
+              <div className="show-all-hide-all-container">
+                <button className="button-clear toggle-all-apps-button show-all-button" 
+                  onClick={() => { this.toggleAllApps(true).bind(this); }}>Show all</button>
```

Comment:
That'll bind the `this` context to the appropriate component and the `true` to the first argument of the function.